AElf.OS.Network.Grpc/GrpcPeerService.cs

```diff
@@ -0,0 +1,191 @@
+using System;
+using System.Threading.Tasks;
+using AElf.Common;
+using AElf.Kernel;
+using AElf.OS.Network.Events;
+using AElf.OS.Network.Grpc.Events;
+using AElf.OS.Network.Temp;
+using Google.Protobuf;
+using Grpc.Core;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Volo.Abp.EventBus.Local;
+
+namespace AElf.OS.Network.Grpc
+{
+    /// <summary>
+    /// Implementation of the grpc generated service. It contains the rpc methods
+    /// exposed to peers.
+    /// </summary>
+    public class GrpcPeerService : PeerService.PeerServiceBase
+    {
+        /// <summary>
+        /// Event launched when a peer disconnects explicitly.
+        /// </summary>
+        public event EventHandler PeerSentDisconnection;
+        
+        private readonly IPeerAuthentificator _peerAuthenticator;
+        private readonly IBlockService _blockService;
+        private readonly ILocalEventBus _localEventBus;
+
+        public ILogger<GrpcPeerService> Logger;
+        
+        public GrpcPeerService(IPeerAuthentificator peerAuthenticator,
+            IBlockService blockService, ILocalEventBus localEventBus)
+        {
+            _peerAuthenticator = peerAuthenticator;
+            _blockService = blockService;
+            _localEventBus = localEventBus;
+
+            Logger = NullLogger<GrpcPeerService>.Instance;
+        }
+
+        /// <summary>
+        /// First step of the connect/auth process.Used to initiate a connection. The provided payload should be the
+        /// clients authentication information. When receiving this call, protocol dictates you send the client your auth
+        /// information. The response says whether or not you can connect.
+        /// </summary>
+        public override Task<AuthResponse> Connect(Handshake request, ServerCallContext context)
+        {
+            Logger?.LogTrace($"[{context.Peer}] has initiated a connection request.");
+            
+            try
+            {
+                var peerServer = context.Peer.Split(":")[1] + ":" + request.HskData.ListeningPort;
```

Comment:
context.Peer.Split(":")[1]

may make a class, Parse(string ) ToString().

do not split everywhere...

Reply:
or you can store it using Url class